pkg/tickers/retry.go

```diff
@@ -0,0 +1,94 @@
+package tickers
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+	"github.com/smartcontractkit/ocr2keepers/pkg/util"
+)
+
+const (
+	retryDelay              = 500 * time.Millisecond
+	totalAttempt            = 3
+	attemptsCacheExpiration = 10 * retryDelay // 5 seconds
+)
+
+type retryTicker struct {
+	timeTicker
+	nextRetries     sync.Map // time.Time -> ocr2keepers.UpkeepPayload
+	payloadAttempts *util.Cache[int]
+}
+
+// Retry adds a retryable result to the retryTicker.
+func (rt *retryTicker) Retry(result ocr2keepers.CheckResult) error {
+	payload := result.Payload
+
+	if result.Retryable {
+		attemptCount, ok := rt.payloadAttempts.Get(payload.ID)
+		if !ok {
+			attemptCount = 0
+		}
+
+		if attemptCount >= totalAttempt {
+			fmt.Printf("Payload %s has already been tried %d times. Skipping.\n", payload.ID, totalAttempt)
+			return nil
+		}
+
+		attemptCount++
+		rt.payloadAttempts.Set(payload.ID, attemptCount, util.DefaultCacheExpiration)
```

Comment:
Do we want the attempt count to expire after 5 seconds? I would have thought we would want to track total retry count over a much longer period?

Reply:
If we dont expire them, this map in memory will explode over time. IIUC, in one `session`, we keep track of only max 3 times of retry for a given payload. Once this payload expires, a new session starts, and we can retry max 3 times for the same payload.